test/suite/run.py

```diff
@@ -651,7 +656,7 @@ def get_sort_keys(test):
                     hugetests.add(name)    # warn for too many scenarios
             return (s, test.simpleName())  # sort by scenario number first
         all_tests = sorted(tests, key = get_sort_keys)
-        if not longtest:
+        if not longtest or extralongtest:
```

Comment:
Do you actually mean `if (not longtest) and (not extralongtest):` or `if not (longest or extralongtest):`?

Reply:
I missed the second not here, so we want `not longtest or not extralongtest`; which I think simplifies to `not(longtest and extralongtest)`.

Reply:
addressed here: [034987c](https://github.com/wiredtiger/wiredtiger/pull/8957/commits/034987c691596cb85c27db0f0fb9229cd4bdf11d)

Reply:
If longtest is true and extralongtest is false (i.e. when "--long" command option is specified), the `not (longtest and extralongtest)` condition will be met, which is undesired?

Reply:
I see what you mean, we don't want to go into the loop if --long is given, or if --extra-long is given, and only when both are not specified. Updated here: [eb1c40d](https://github.com/wiredtiger/wiredtiger/pull/8957/commits/eb1c40d5a6af620ee1a1ac016f52ab007d49561c)